src/main/php/lang/ast/emit/Reflection.class.php

```diff
@@ -23,6 +24,31 @@ public function __construct($type) {
   /** @return string */
   public function name() { return $this->reflect->name; }
 
+  /** @return iterable */
+  public function implementedInterfaces() { return $this->type->getInterfaceNames(); }
+
+  /**
+   * Checks whether a given method exists
+   *
+   * @param  string $named
+   * @return bool
+   */
+  public function providesMethod($named) {
+    return $this->reflect->hasMethod($named);
+  }
+
+  /**
+   * Returns all methods annotated with a given annotation
+   *
+   * @param  string $annotation
+   * @return iterable
+   */
+  public function methodsAnnotated($annotation) {
+    foreach ((new Reflect($this->reflect))->methods()->annotated($annotation) as $method) {
+      yield $method->name() => $this->reflect->getMethod($method->name())->getStartLine();
```

Comment:
The reflection API should expose the start and end lines as well as declaring files, e.g. via `declaredIn()`, which would rerurn an object encapsulating this information.

Reply:
Alternatively, we should expose the meta information accessors so we can work with this low level API